### ISSUE TITLE
applying materialboxed to IMG within carousel-item makes it disappear after zoom on click

### ISSUE THREAD

**Issue by badpenguin** [Dogfalo/materialize#2597]
$('.carousel').carousel({dist:0}); is great!
But if i add the materialboxed to the img inside the carousel-item when i click it the image is zoomed
the screen becomes dark grey
but when image's zoom animation ends the image is hidden
and i'm left with just this dark background.

```
<div class="carousel">
    <li class="carousel-item" href="#one!"><img class="materialboxed" src="http://lorempixel.com/250/250/nature/1">
    <li class="carousel-item" href="#two!"><img class="materialboxed" src="http://lorempixel.com/250/250/nature/2">
    <li class="carousel-item" href="#three!"><img class="materialboxed" src="http://lorempixel.com/250/250/nature/3">
    <li class="carousel-item" href="#four!"><img class="materialboxed" src="http://lorempixel.com/250/250/nature/4">
    <li class="carousel-item" href="#five!"><img class="materialboxed" src="http://lorempixel.com/250/250/nature/5">
  </div>

<div class="carousel carousel-slider">
    <a class="carousel-item" href="#one!"><img class="materialboxed" src="http://lorempixel.com/800/400/food/1"></a>
    <a class="carousel-item" href="#two!"><img src="http://lorempixel.com/800/400/food/2"></a>
    <a class="carousel-item" href="#three!"><img src="http://lorempixel.com/800/400/food/3"></a>
    <a class="carousel-item" href="#four!"><img src="http://lorempixel.com/800/400/food/4"></a>
  </div>
```

**Comment by Umar3x** [comment]
I've got the exact same probleme. We can't set a width for images in that carousel. That could be very nice ! Btw I confirm badpenguin, materialboxed renders a zoom and then a dark grey screen ... It seems that its the final rendering of the img that crashes. 
...

**Comment by CassianoSumer** [comment]
+1
Messed around with the z-index of my elements and it doesn't seem right. When I set the z-index of the overlay to -10 to test it, I could see the photo, but there were still some elements above it, which doesn't seem to be right.

I tried some workaround, creating a modal that shows the larger picture, and it works fine, except on mobile. I also tried to use swipebox instead of materialbox (the gallery feature was nice to have) and had the same issue.
The <a class="carousel-item"> seems to be unclickable on mobile (I confirmed this on http://materializecss.com/media.html, clicking the photos at the carousel doesn't change the url to http://materializecss.com/media.html#four!, for example). Is this supposed to be expected behavior?

**Comment by qantumthemes** [comment]
I can confirm the issue, link events in mobile, for links contained in the carousel, are not executed. Not even in real mobile devices, galaxy S5 or iPhone 6s and 6s+.
Both in Safari and Chrome, while Firefox works.

Looks like there is some event delegation blocking the click behaviour, but can't change this as is in the materializeCSS core. Please fix :) I love this component!

**Comment by patrickonparker** [comment]
This is driving me insane! I am using a carousel in my portfolio and links do nothing!

**Comment by davidalejandroaguilar** [comment]
Me too but this happens when I put images on a Masonry-like grid as in this codepen: http://codepen.io/anon/pen/AXzpba. 

Yes, I'm using the javascript initialization:

```
$(document).ready(function(){
    $('.materialboxed').materialbox();
  });
```

**Comment by acburst** [comment]
Fixed in 428d7fe

**Comment by jaimeplier** [comment]
I'm having the same problem when adding materialboxed class to an image within a slider, any idea how to solve this?
